### PR TITLE
Issue 7069 - Add Subnet/CIDR Support for HAProxy Trusted IPs

### LINKED ISSUE #7069: Add Subnet/CIDR Support for HAProxy Trusted IPs
**Is your feature request related to a problem? Please describe.**
Right now `nsslapd-haproxy-trusted-ip` only accepts individual IP addresses. When you need to trust a whole subnet of HAProxy instances (like an entire load balancer pool or internal network), you have to manually add every single IP. For a /24 that's 254 entries. It's tedious and annoying to manage.

**Describe the solution you'd like**
Support CIDR notation in nsslapd-haproxy-trusted-ip so we can do:

- 192.168.0.0/24 for IPv4 subnets
- 2001:db8::/32 for IPv6 subnets
- Mix individual IPs and subnets in the same config

**Describe alternatives you've considered**
- Adding IPs manually (current method, doesn't scale)
- Turning off IP validation entirely (bad idea security-wise)

**Additional context**
Pretty standard feature for enterprise setups where HAProxy instances live in dedicated network ranges. CIDR notation is the standard way to define subnets and more precise than wildcards.

### PR DESCRIPTION
**Pull request by droideck**
Description: nsslapd-haproxy-trusted-ip now supports CIDR notation (192.168.0.0/24, 2001:db8::/32) instead of requiring individual IPs for each address in a subnet. This makes it practical to trust entire HAProxy network ranges without manually adding hundreds of entries.

The implementation includes CIDR parsing with validation, netmask precomputation for performance, and support for mixing individual IPs and subnets. Added comprehensive tests for subnet matching, edge cases, and malformed input validation. Updated Cockpit console UI accordingly.

Fixes: https://github.com/389ds/389-ds-base/issues/7069

Reviewed by: ?

## Summary by Sourcery

Add support for CIDR-based subnets to the HAProxy trusted IP feature, including parsing, validation, and optimized runtime matching; update configuration handling, UI, CLI, and tests to accommodate both individual IPs and subnets.

New Features:
- Allow nsslapd-haproxy-trusted-ip to accept CIDR notation for IPv4 and IPv6 subnets alongside individual IPs
- Expose binary parsed entries via g_get_haproxy_trusted_ip_parsed for efficient connection validation

Enhancements:
- Precompute and store network/netmask pairs for each trusted entry to speed up IP matching
- Switch runtime HAProxy header validation to use binary subnet matching instead of string comparisons
- Enforce stricter validation in config_set_haproxy_trusted_ip to reject malformed inputs (wildcards, extra slashes, whitespace, etc.)
- Refactor memory allocations to use slapi_ch_malloc and integrate parsed entries into frontend config

Documentation:
- Update Cockpit console validation and labels to accept and describe IP/subnet CIDR notation

Tests:
- Add cmocka unit tests for IPv4/IPv6 subnet logic, CIDR matching, netmask precomputation, and edge cases
- Add Python integration tests to cover valid and invalid CIDR configuration scenarios in server and dsconf CLI
- Update existing haproxy header parsing and CLI test suites to include subnet coverage

Chores:
- Bump copyright years across multiple source and test files to 2025

### CODE REVIEW

File: ldap/servers/slapd/connection.c [hunk 1]
```diff
@@ -1332,36 +1332,45 @@ connection_read_operation(Connection *conn, Operation *op, ber_tag_t *tag, int *
...
+                            trusted_matches_destip_found = haproxy_ip_matches_parsed(&pr_netaddr_dest,
+                                                                                     parsed_entries,
+                                                                                     parsed_count);
+                        } else {
+                            /* Parsed entries should always be available after config load.
+                             * If they're not, this indicates a configuration initialization failure. */
+                            slapi_log_err(SLAPI_LOG_ERR, "connection_read_operation",
+                                         "HAProxy trusted IPs not properly initialized - disconnecting connection\n");
+                            disconnect_server_nomutex(conn, conn->c_connid, -1,
+                                                     SLAPD_DISCONNECT_PROXY_UNKNOWN, EPROTO);
```
Comment (mreynolds389) [inline ldap/servers/slapd/connection.c]: Indentation in two spots above

File: ldap/servers/slapd/haproxy.c [hunk 2]
```diff
@@ -1,392 +1,874 @@
...
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+#include <fcntl.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include "slap.h"
+
+/* Function to parse IPv4 addresses in version 2 */
+static int haproxy_parse_v2_addr_v4(uint32_t in_addr, unsigned in_port, PRNetAddr *pr_netaddr)
```
Comment (mreynolds389) [inline ldap/servers/slapd/haproxy.c]: Not a big deal, we usually keep the function name and params on a newline versus the entire declaration all in one line

File: ldap/servers/slapd/result.c [hunk 3]
```diff
@@ -199,15 +211,72 @@ g_set_haproxy_trusted_ip(struct berval **ipaddress)
...
+    /*
+     * Parsing must succeed. If it fails, this indicates validation in config_set_haproxy_trusted_ip
+     * didn't catch an issue, or there's a logic error. This should never happen in production.
+     */
+    if (parsed_entries == NULL && parsed_count == 0 && ipaddress && ipaddress[0]) {
+        slapi_log_err(SLAPI_LOG_ERR, "g_set_haproxy_trusted_ip",
+                     "CRITICAL: Failed to parse trusted IPs into binary format. "
+                     "This should have been caught during validation.\n");
+        /* Free the temporary array and return without updating config */
+        for (int i = 0; haproxy_trusted_ip[i] != NULL; i++) {
```
Comment (mreynolds389) [inline ldap/servers/slapd/result.c]: Use size_t for "for" loops please

**Review by mreynolds389 — APPROVED** [review]
Ack!
